### PR TITLE
Fixed neutral actors being visible in the All Players observer shroud mode

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/8961.

### CODE REVIEW

File: OpenRA.Game/Player.cs [hunk 1]
```diff
@@ -137,8 +137,8 @@ public override string ToString()
 		public Dictionary<Player, Stance> Stances = new Dictionary<Player, Stance>();
 		public bool IsAlliedWith(Player p)
 		{
-			// Observers are considered as allies
-			return p == null || Stances[p] == Stance.Ally || p.Spectating;
+			// Human observers are considered as allies
```
Comment (pchote) [inline OpenRA.Game/Player.cs]: This isn't quite right.  You're attributing humanity to the wrong player, but then that's confusing with respect to the AI...

How about `// Observers are considered allies to active combatants`

Comment (Mailaender) [inline OpenRA.Game/Player.cs]: Hadn't thought about the bots. Updated.

**Comment by pchote** [comment]
Note for testing: the Spectating flag is there mainly for players who die, so make sure to test this case and not just regular observers.

**Comment by penev92** [comment]
Works :+1:

**Comment by abcdefg30** [comment]
:+1: / :white_check_mark:

**Pull request merged**